lib/Data/ContentTypeData.php

```diff
@@ -44,4 +44,25 @@ public function addFieldDefinitionData(FieldDefinitionData $fieldDefinitionData)
     {
         $this->fieldDefinitionsData[] = $fieldDefinitionData;
     }
+
+    /**
+     * Sort $this->fieldDefinitionsData first by position, then by creation order.
```

Comment:
@pspanja Yep, agreed. CC @andrerom (just to clarify, `priority` would still be the first sort criterion)